### PR TITLE
Routes and navigators in flutter apps

### LINKED ISSUE #4092: Introduction of Routing and navigation in Flutter Applications
#### Pre-submission advice
By following all our pre-submission advice and reviewing our Resources folder, you will maximise your chances of your topic being approved.

We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner.

Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles that will help you propose and write a successful article.

**Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

#### Introduction of Routing and navigation in Flutter
Your title should be descriptive of the article/tutorial. Be Specific. 

Use keyword research to improve your article's SEO.

#### Proposed article introduction
Navigation in a small application is not that hard you only need to use the standard navigation that flutter comes packaged with. But when your project grows you’ll want to add authentication, authorizations, dynamic links with all of this you’d probably want less time thinking how navigation is going to work with all of this and spend more time building your business logic. GetX does a good job of helping you do this using middlewares and bindings, child routes.

Mobile apps typically display full-screen elements called “screens” or “pages”. In Flutter these elements are called routes and they’re managed by a Navigator widget. The navigator manages a stack of Route objects and provides methods for managing the stack, like Navigator.push and Navigator.pop.

For example, if the article is based on machine learning, use the following: [Machine learning] Introduction to Machine Learning.
If the article is based on developing an Android application, use the following: [Android] Developing Apps using Android.

ONLY viable topics to pick from:
- API
- Artificial Intelligence
- Blockchain
- Containers
- Edge Computing
- Languages
- Machine Learning
- Networking
- Node.js
- Security

#### Key takeaways 
- learn how to use various navigator and routes.
- how to pass data between pages
-  how these APIs enable more fine-tuned control over the screens in your app.
- Build a simple flutter application
- how you can use Navigator 2.0 and Router to parse routes.
#### Article quality
How is your article unique? Tell us what makes your approach different from similar articles that have published on the same topic?
Is yours more in-depth? Does it cover additional topics? Do you provide handy tips or anecdotal advice?

#### References
- https://api.flutter.dev/flutter/widgets/Navigator-class.html
- https://medium.com/flutter-community/flutter-routes-and-navigation-69f128a9ea8f
- https://medium.com/flutter/learning-flutters-new-navigation-and-routing-system-7c9068155ade

### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by karehnikita**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

`This closes #4092

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by karehnikita** [comment]
@srishilesh @hectorkambow @WanjaMIKE I created a new PR and seems okay. May we proceed kindly.

**Comment by srishilesh** [comment]
@karehnikita 

Please wait until a peer-reviewer picks up this PR to review it. I've already works in my backlog.

**Comment by srishilesh** [comment]
> Hey @srishilesh 👋 
> Do I proceed with the review?
> 
> cc @karehnikita 

Hey @BrianaNzivu ,

Sure. Please go ahead with the review.

Thanks.

**Comment by Kayere** [comment]
Hello @karehnikita,
 Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept.

Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content.

Be sure to see our resources page to see more info on plagiarism and what is considered as such.

[Routes and navigators in flutter apps #5905 Copyleaks Report.pdf](https://github.com/section-engineering-education/engineering-education/files/7822452/Routes.and.navigators.in.flutter.apps.5905.Copyleaks.Report.pdf)

**Comment by karehnikita** [comment]
@Kayere Hello thanks for your response to my PR . I have checked on the plagiarism and any issue that arises are flutter keywords or default methods. Thank you for the feedback

**Review by BrianaNzivu — CHANGES_REQUESTED** [review]
Hey @karehnikita 👋 .

First of all, welcome to Section - EngEd. :tada: 
We are happy to have you here.

My name is Briana, I'll be reviewing your article for publishing. I will make this review as fast and as swift as possible, with your help of course 😄 .

Good job on the article. It is a very interesting read. 

Just a few changes I need you to make as we continue with the preliminary revisions:
- Please structure your article according to this [sample](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md). I need you to clearly highlight and explain your steps. Where you can add more detail please do so.
- Your avatar image should be a headshot image of yourself. Please correct that.  Ensure it is 600 x 400 px as required. You can use [Befunky](https://www.befunky.com/) to do this.
- I want you to cross-check your article again using [Grammarly](https://app.grammarly.com/). Please correct any grammar mistakes and hard-to-read sentences.

In case of any question or help needed please do not hesitate to ping me here or on Slack.
Also, notify me when you are done both here and on Slack.

Regards.

**Comment by BrianaNzivu** [comment]
Hey @karehnikita 👋.
Just checking up on you if everything is ok. 
Do you need some help?

**Comment by karehnikita** [comment]
almost done @BrianaNzivu

File: content/articles/routes-and-navigators-in-flutter-apps/index.md [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+### Simple application that uses navigators and routes.
+
+For our application, we will have 3 screens. 
+
+### Step one: Create a new flutter application
+  Open your visual code. On the menu bar click on `view` then `command palette` then click on `new flutter project`. Specify the location where to create your project then specify the name of your project. You can also create using the vs code terminal by typing 
+```cmd
+flutter create name_of_project 
+```
+### Step two: Named routes
```
Comment (BrianaNzivu) [inline content/articles/routes-and-navigators-in-flutter-apps/index.md]: This paragraph is not clear.

File: content/articles/routes-and-navigators-in-flutter-apps/index.md [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+ FirstScreen.routeName: (context) => FirstScreen(),
+ Secondscreen.routeName: (context) => const Secondscreen(),
+ Thirdscreen.routeName: (context) => const Thirdscreen(),
+      },
+    );
+  }
+}
+
+```
+### Step three: Coding firstpage.
```
Comment (BrianaNzivu) [inline content/articles/routes-and-navigators-in-flutter-apps/index.md]: You can give better titles for all the steps. Make them a little bit more descreptive.

File: content/articles/routes-and-navigators-in-flutter-apps/index.md [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+```
+### Step 6. Create allfiles.dart. It contains all the exports of all other files. File used as one import to reduce the repetition of imports.
```
Comment (BrianaNzivu) [inline content/articles/routes-and-navigators-in-flutter-apps/index.md]: Describe this step well.

File: content/articles/routes-and-navigators-in-flutter-apps/index.md [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+![FirstPage](/engineering-education/introduction-of-routing-and-navigation-in-flutter-applications/firstpage.jpg)
+
+![SecondPage](/engineering-education/introduction-of-routing-and-navigation-in-flutter-applications/secondpage.jpg)
+
+
+
+![ThirdPage](/engineering-education/introduction-of-routing-and-navigation-in-flutter-applications/thirdpage.jpg)
+
+
+For any query Reachout @ [this GitHub repository](https://github.com/karehnikita/).
```
Comment (BrianaNzivu) [inline content/articles/routes-and-navigators-in-flutter-apps/index.md:275]: Let's have the repo with this project's code and a sample APK to this project instead.

File: content/articles/routes-and-navigators-in-flutter-apps/index.md [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+
+### Conclusion
+
```
Comment (BrianaNzivu) [inline content/articles/routes-and-navigators-in-flutter-apps/index.md:278]: The conclusion summarizes what the article has covered and the very key points. So you can fix that too.

**Review by BrianaNzivu — CHANGES_REQUESTED** [review]
Hey @karehnikita 👋 

Your PR is still quite a bit away from being completed.  I have left some comments on what needs changes. 
**Please ensure you make changes to the updated `index.md`  file.**
 
In an effort to be fair to our larger EngEd community as a rule of thumb - we can only accept content (pull requests) that are complete and fully polished articles/tutorials - meaning they are ready to be published. 
 
All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) - this allows all revisions to be done on your forked personal repo. 
 
I will be closing this PR and please re-submit it when it is ready to be published as a new PR (and you can add the duplicate label) and ping me both on Github and Slack so that I can review your article. 
 
>NOTE: A duplicate PR will go back into the queue and will not receive any further major revisions. A duplicate PR should be 100% ready to be published, otherwise, it will be closed immediately. We encourage our new contributors to go over our [resources page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) and [new contributor folder](https://github.com/section-engineering-education/engineering-education/tree/master/new_contributors) to be better familiar with the level of quality and style we are looking to publish.
 
Closing a PR does NOT mean the article can not be published (or fixed) it simply means that the edits, revisions, and fixes will happen outside of the queue.
 
Do not hesitate to reach out in case of any clarification or problem encountered.

Regards.

**Pull request closed without merging**